NoClassDefFoundError: com.ibm.streamsx.rabbitmq.i18n.Messages

I cloned the RabbitMQ repo and built the toolkit. I then attempted to run the sample, however I used a wrong host/port combo which resulted in a connection error. However, in addition to the connection error, the following exception was reported:

```
Exception in thread "AMQP Connection 9.30.230.179:25672" java.lang.ExceptionInInitializerError
	at java.lang.J9VMInternals.ensureError(J9VMInternals.java:141)
	at java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:130)
	at com.ibm.streamsx.rabbitmq.RabbitMQConnectionExceptionHandler.handleUnexpectedConnectionDriverException(RabbitMQConnectionExceptionHandler.java:74)
	at com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:575)
	at java.lang.Thread.run(Thread.java:785)
Caused by: java.util.MissingResourceException: Can't find bundle for base name com.ibm.streamsx.rabbitmq.i18n.CommonMessages, locale en_US
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1590)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1406)
	at java.util.ResourceBundle.getBundle(ResourceBundle.java:784)
	at com.ibm.streamsx.rabbitmq.i18n.Messages.<clinit>(Messages.java:15)
	... 3 more
06 Sep 2017 07:06:51.124 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:com.ibm.streams.operator.internal.jni.JNIBridge.process:-1]  - com.ibm.streamsx.rabbitmq.i18n.Messages (initialization failure)
06 Sep 2017 07:06:51.125 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - java.lang.NoClassDefFoundError: com.ibm.streamsx.rabbitmq.i18n.Messages (initialization failure)
06 Sep 2017 07:06:51.125 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:91)
06 Sep 2017 07:06:51.126 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQBaseOper.initializeRabbitChannelAndConnection(RabbitMQBaseOper.java:187)
06 Sep 2017 07:06:51.126 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQSink.process(RabbitMQSink.java:67)
06 Sep 2017 07:06:51.126 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.runtime.api.OperatorAdapter.processTuple(OperatorAdapter.java:1591)
06 Sep 2017 07:06:51.126 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.runtime.api.OperatorAdapter$1.tuple(OperatorAdapter.java:825)
06 Sep 2017 07:06:51.127 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.runtime.api.OperatorAdapter$1.tuple(OperatorAdapter.java:818)
06 Sep 2017 07:06:51.127 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.ports.SwitchingHandler$Alternate.tuple(SwitchingHandler.java:162)
06 Sep 2017 07:06:51.127 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.network.DeserializingStream.tuple(DeserializingStream.java:76)
06 Sep 2017 07:06:51.128 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.network.DeserializingStream.processRawTuple(DeserializingStream.java:65)
06 Sep 2017 07:06:51.128 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.runtime.api.InputPortsConduit.processRawTuple(InputPortsConduit.java:100)
06 Sep 2017 07:06:51.128 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.jni.JNIBridge.process(JNIBridge.java:312)
06 Sep 2017 07:06:51.129 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:com.ibm.streams.operator.internal.jni.JNIBridge.complete:-1]  - 
Exception in thread "Thread-21" java.lang.NoClassDefFoundError: com.ibm.streamsx.rabbitmq.i18n.Messages (initialization failure)
06 Sep 2017 07:06:51.130 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - java.lang.ExceptionInInitializerError
	at java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:91)
	at com.ibm.streamsx.rabbitmq.RabbitMQBaseOper.initializeRabbitChannelAndConnection(RabbitMQBaseOper.java:187)
06 Sep 2017 07:06:51.130 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - java.lang.J9VMInternals.ensureError(J9VMInternals.java:141)
	at com.ibm.streamsx.rabbitmq.RabbitMQSink.process(RabbitMQSink.java:67)
	at com.ibm.streams.operator.internal.runtime.api.OperatorAdapter.processTuple(OperatorAdapter.java:1591)06 Sep 2017 07:06:51.130 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:130)

	at com.ibm.streams.operator.internal.runtime.api.OperatorAdapter$1.tuple(OperatorAdapter.java:825)06 Sep 2017 07:06:51.130 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQBaseOper.initializeRabbitChannelAndConnection(RabbitMQBaseOper.java:187)

	at com.ibm.streams.operator.internal.runtime.api.OperatorAdapter$1.tuple(OperatorAdapter.java:818)
06 Sep 2017 07:06:51.131 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQSource.produceTuples(RabbitMQSource.java:179)
	at com.ibm.streams.operator.internal.ports.SwitchingHandler$Alternate.tuple(SwitchingHandler.java:162)
06 Sep 2017 07:06:51.131 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQSource.access$000(RabbitMQSource.java:46)	at com.ibm.streams.operator.internal.network.DeserializingStream.tuple(DeserializingStream.java:76)

	at com.ibm.streams.operator.internal.network.DeserializingStream.processRawTuple(DeserializingStream.java:65)
06 Sep 2017 07:06:51.131 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQSource$1.run(RabbitMQSource.java:93)
	at com.ibm.streams.operator.internal.runtime.api.InputPortsConduit.processRawTuple(InputPortsConduit.java:100)
06 Sep 2017 07:06:51.131 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - java.lang.Thread.run(Thread.java:785)
	at com.ibm.streams.operator.internal.jni.JNIBridge.process(JNIBridge.java:312)
Caused by: 06 Sep 2017 07:06:51.132 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_javaop M[?:?:0]  - com.ibm.streams.operator.internal.runtime.OperatorThreadFactory$2.run(OperatorThreadFactory.java:137)
Exception in thread "Thread-22" java.util.MissingResourceException: Can't find bundle for base name com.ibm.streamsx.rabbitmq.i18n.CommonMessages, locale en_US
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1590)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1406)06 Sep 2017 07:06:51.132 [128596] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:com.ibm.streams.operator.internal.runtime.api.OperatorAdapter.backgroundException:-1]  - com.ibm.streamsx.rabbitmq.i18n.Messages (initialization failure)

	at java.util.ResourceBundle.getBundle(ResourceBundle.java:784)
06 Sep 2017 07:06:51.132 [128596] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - java.lang.NoClassDefFoundError: com.ibm.streamsx.rabbitmq.i18n.Messages (initialization failure)
	at com.ibm.streamsx.rabbitmq.i18n.Messages.<clinit>(Messages.java:15)
06 Sep 2017 07:06:51.132 [128596] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:91)
	at com.ibm.streamsx.rabbitmq.RabbitMQConnectionExceptionHandler.handleUnexpectedConnectionDriverException(RabbitMQConnectionExceptionHandler.java:74)
	at com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:575)06 Sep 2017 07:06:51.132 [128596] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.ibm.streamsx.rabbitmq.RabbitMQConnectionExceptionHandler.handleUnexpectedConnectionDriverException(RabbitMQConnectionExceptionHandler.java:74)

	at java.lang.Thread.run(Thread.java:785)
06 Sep 2017 07:06:51.133 [128596] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:575)
06 Sep 2017 07:06:51.133 [128596] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_javaop M[?:?:0]  - java.lang.Thread.run(Thread.java:785)
java.lang.ExceptionInInitializerError
	at java.lang.J9VMInternals.ensureError(J9VMInternals.java:141)
	at java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:130)
	at com.ibm.streamsx.rabbitmq.RabbitMQBaseOper.initializeRabbitChannelAndConnection(RabbitMQBaseOper.java:187)
	at com.ibm.streamsx.rabbitmq.RabbitMQSource.produceTuples(RabbitMQSource.java:179)
	at com.ibm.streamsx.rabbitmq.RabbitMQSource.access$000(RabbitMQSource.java:46)
	at com.ibm.streamsx.rabbitmq.RabbitMQSource$1.run(RabbitMQSource.java:93)
	at java.lang.Thread.run(Thread.java:785)
	at com.ibm.streams.operator.internal.runtime.OperatorThreadFactory$2.run(OperatorThreadFactory.java:137)
Caused by: java.util.MissingResourceException: Can't find bundle for base name com.ibm.streamsx.rabbitmq.i18n.CommonMessages, locale en_US
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1590)
06 Sep 2017 07:06:51.134 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,spl_function M[JavaOp.cpp:processTuple:1568]  - CDISR5126E: A Java exception occurred at Operator.process() during the JavaOp invocation.	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1406)

	at java.util.ResourceBundle.getBundle(ResourceBundle.java:784)
	at com.ibm.streamsx.rabbitmq.i18n.Messages.<clinit>(Messages.java:15)
	... 6 more
06 Sep 2017 07:06:51.135 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_function M[JavaOp.cpp:complete:1438]  - CDISR5126E: A Java exception occurred at Operator.allPorts() asynchronous during the JavaOp invocation.
06 Sep 2017 07:06:51.136 [128580] ERROR #splapplog,J[0],P[0],RabbitMQStream,spl_pe M[PEImpl.cpp:logTerminatingException:1893]  - CDISR5033E: An exception occurred during the execution of the RabbitMQStream operator. Processing element number 0 is terminating.
06 Sep 2017 07:06:51.136 [128580] ERROR #splapptrc,J[0],P[0],RabbitMQStream,#splapptrc,RabbitMQStream,spl_operator M[OperatorThread.cpp:run:82]  - CDISR5030E: An exception occurred during the execution of the RabbitMQStream operator. The exception is: A Java exception occurred at Operator.allPorts() asynchronous during the JavaOp invocation.
06 Sep 2017 07:06:51.137 [128577] ERROR #splapplog,J[0],P[0],RabbitMQSinkOp,spl_pe M[PEImpl.cpp:logTerminatingException:1893]  - CDISR5033E: An exception occurred during the execution of the RabbitMQSinkOp operator. Processing element number 0 is terminating.
06 Sep 2017 07:06:51.137 [128577] ERROR #splapptrc,J[0],P[0],RabbitMQSinkOp,#splapptrc,RabbitMQSinkOp,spl_operator M[OperatorThread.cpp:run:82]  - CDISR5030E: An exception occurred during the execution of the RabbitMQSinkOp operator. The exception is: A Java exception occurred at Operator.process() during the JavaOp invocation.
06 Sep 2017 07:06:51.145 [128326] ERROR #splapptrc,J[0],P[0],RabbitMQStream,spl_operator M[PEImpl.cpp:process:1301]  - CDISR5053E: Runtime failures occurred in the following operators: RabbitMQSinkOp, RabbitMQStream.

```

--- a/com.ibm.streamsx.rabbitmq/impl/java/src/com/ibm/streamsx/rabbitmq/i18n/Messages.java
+++ b/com.ibm.streamsx.rabbitmq/impl/java/src/com/ibm/streamsx/rabbitmq/i18n/Messages.java
@@ -9,7 +9,7 @@ import java.util.MissingResourceException;
 import java.util.ResourceBundle;
 
 public class Messages {
-	private static final String BUNDLE_NAME = "com.ibm.streamsx.rabbitmq.i18n.CommonMessages"; //$NON-NLS-1$
+	private static final String BUNDLE_NAME = "com.ibm.streamsx.rabbitmq.i18n.Messages"; //$NON-NLS-1$
 
 	private static final ResourceBundle RESOURCE_BUNDLE = ResourceBundle
 			.getBundle(BUNDLE_NAME);

--- a/com.ibm.streamsx.rabbitmq/info.xml
+++ b/com.ibm.streamsx.rabbitmq/info.xml
@@ -50,7 +50,7 @@ Alternatively, you can fully qualify the operators that are provided by toolkit 
 7. Run the application. You can submit the application as a job by using the **streamtool submitjob** command or by using Streams Studio. 
 
 	</info:description>
-    <info:version>1.0.0</info:version>
+    <info:version>1.0.1</info:version>
     <info:requiredProductVersion>4.2.0.0</info:requiredProductVersion>
   </info:identity>
   <info:dependencies/>

--- a/samples/RabbitMQSample/Makefile
+++ b/samples/RabbitMQSample/Makefile
@@ -2,9 +2,31 @@
 # others. All Rights Reserved.                        
 .PHONY: all clean
 
-TOOLKITS ?= ../../com.ibm.streamsx.rabbitmq
+SPLC_FLAGS = -a
+ifeq ($(STREAMS_STUDIO_BUILDING), 1)
+  # Building from Streams Studio, use env vars set by studio
+  SPLC = $(STREAMS_STUDIO_SC_PATH)
+  SPL_PATH = $(STREAMS_STUDIO_SPL_PATH)
+  OUTPUT_DIR = $(STREAMS_STUDIO_OUTPUT_DIRECTORY)
+else
+  # Building outside of Streams Studio...
+  ifeq ($(STREAMS_INSTALL),)
+    $(error error: environment variable STREAMS_INSTALL has to be set)
+  endif
+  # Set the STREAMSX_RABBITMQ_TOOLKIT variable if streamsx.rabbitmq sample or toolkit location changed in your environment.
+  ifeq ($(STREAMSX_RABBITMQ_TOOLKIT),)
+    ifneq ($(realpath ../../com.ibm.streamsx.rabbitmq),)
+      STREAMSX_RABBITMQ_TOOLKIT = ../../com.ibm.streamsx.rabbitmq
+    endif
+    ifneq ($(realpath $(STREAMS_INSTALL)/toolkits/com.ibm.streamsx.rabbitmq),)
+      STREAMSX_RABBITMQ_TOOLKIT = $(STREAMS_INSTALL)/toolkits/com.ibm.streamsx.rabbitmq
+    endif
+  endif
+  SPLC = $(STREAMS_INSTALL)/bin/sc
+  SPL_PATH = $(STREAMSX_RABBITMQ_TOOLKIT)
+endif
 
-SPLC_FLAGS ?= -a -t ${TOOLKITS}
+SPLC_FLAGS += -t $(SPL_PATH)
 
 SPLC = $(STREAMS_INSTALL)/bin/sc
 

